src/distribute_kv_cache/rpc/workerpool.rs

```diff
@@ -0,0 +1,223 @@
+use std::{
+    fmt::{self, Debug},
+    sync::Arc,
+};
+
+use async_trait::async_trait;
+use tracing::debug;
+
+use super::error::RpcError;
+
+/// A job that can be executed by a worker.
+type JobImpl = Box<dyn Job + Send + Sync + 'static>;
+
+/// A worker that can execute async jobs.
+#[derive(Clone)]
+pub struct WorkerPool {
+    /// The number of workers in the worker pool.
+    /// Current implementation is that the worker pool with a fixed number of workers.
+    /// TODO: Test if we need a dynamic worker pool.
```

Comment:
what do you mean dynamic worker pool here?

Reply:
OK, I have updated it.